### PR TITLE
TEL-4617: Allow disable cdr for 401 or 407

### PR DESCRIPTION
**Pull request by dev-ryanc**

### CODE REVIEW

**Review by davidcsi — APPROVED** [review]

File: src/mod/endpoints/mod_sofia/sofia.c [hunk 1]
```diff
@@ -2538,6 +2538,8 @@ void sofia_event_callback(nua_event_t event,
 						switch_log_printf(SWITCH_CHANNEL_LOG, SWITCH_LOG_DEBUG, "detaching session %s\n", sofia_private->uuid);
 						if (switch_true(telnyx_force_cdr)) {
 							switch_channel_clear_flag(channel, CF_NO_CDR);
+						} else if (switch_false(telnyx_force_cdr)) {
```
Comment (pchero) [inline src/mod/endpoints/mod_sofia/sofia.c:2541]: Just question.

Who is setting the `telnyx_force_cdr` channel variable? dialplan? 
Is the dialplan doesn't set the `telnyx_force_cdr` channel variable when it get's 401/407?

Comment (pchero) [reply src/mod/endpoints/mod_sofia/sofia.c:2541]: And how about `else`? not `else if`.

Comment (dev-ryanc) [reply src/mod/endpoints/mod_sofia/sofia.c:2541]: Dialplan is setting this variable. I am enforcing that they have to explicitly set the telnyx_force_cdr value before we decide if we will disable cdr or not. If they don't set the variable we don't do nothing and use the default behavior

Comment (pchero) [reply src/mod/endpoints/mod_sofia/sofia.c:2541]: Thanks, then this needs to be as `else if`. Clear now. :)

**Review by pchero — APPROVED** [review]
Looks OK to me. :)

**Review by StefanYohansson — APPROVED** [review]

**Pull request merged**